src/query/src/sql.rs

```diff
@@ -157,20 +151,46 @@ pub async fn show_tables(
     // TODO(dennis): Specify the order of the results in schema provider API
     tables.sort();
 
-    let tables = if let ShowKind::Like(ident) = stmt.kind {
-        Helper::like_utf8(tables, &ident.value).context(error::VectorComputationSnafu)?
-    } else {
-        Arc::new(StringVector::from(tables))
-    };
+    match stmt.kind {
+        ShowKind::All => {
+            let schema = Arc::new(Schema::new(vec![ColumnSchema::new(
+                TABLES_COLUMN,
+                ConcreteDataType::string_datatype(),
+                false,
+            )]));
+            let columns = vec![Arc::new(StringVector::from(tables)) as _];
+            let record_batch =
+                RecordBatch::new(schema, columns).context(error::CreateRecordBatchSnafu)?;
+            let result = execute_show_with_filter(record_batch, None, "tables".to_string()).await?;
```

Comment:
For `ShowKind::All`, can we just use the old `let records = RecordBatches::try_from_columns(schema, vec![tables]).context(error::CreateRecordBatchSnafu)?;`, instead of going through the whole datafusion execution in `execute_show_with_filter`?